iModelCore/iModelPlatform/DgnCore/DgnUnits.cpp

```diff
@@ -476,9 +476,12 @@ void DgnGeoLocation::SetProjectExtents(AxisAlignedBox3dCR newExtents)
     // JsonCpp and RapidJson differ slightly in precision of floating point numbers.
     // Tile content Ids include a hash of the project extents.
     // Differing precision => different content Ids => invalidate every cached tile in existence.
-    Json::Value jsonObj;
-    BeJsGeomUtils::DRange3dToJson(jsonObj, m_extent);
-    m_dgndb.SavePropertyString(DgnProjectProperty::Extents(), jsonObj.ToString());
+    if (!fromChangesetAppliedEvent)

```

Comment:
Don't see why this check is necessary either. You are expecting this to get called by JsInterop::UpdateProjectExtents only after it's verified that the value is really changed right? I don't like this "fromChangesetAppliedEvent" arg at all...I think it's hacky...

Reply:
The goal is to update the open iModel's props. If its from a changesetAppliedEvent that means the extents were already written to the be_prop table due to the changeset being applied so I figured I should skip that operation.

Reply:
Doesn't make sense to me to call SetProjectExtents with an arg that says don't set the project extents. If you know the project extents changed and all it is that you want is to send out the onProjectExtentsChanged notification you should just do that and not be calling SetProjectExtents. I think Affan's comment on your itwinjs PR is the important one here...

Reply:
switched to affan's suggestion